optimade/models/structures.py

```diff
@@ -751,6 +757,71 @@ class StructureResourceAttributes(EntryResourceAttributes):
         queryable=SupportLevel.MUST,
     )
 
+    @validator("chemical_formula_reduced", "chemical_formula_hill")
+    def check_ordered_formula(cls, v, field):
+        numbers = re.findall(r"(\d+\.?)+", v)
+        elements = re.findall(r"[A-Z][a-z]?", v)
+        expected_elements = sorted(elements)
+
+        if field.name == "chemical_formula_hill":
+            for elem in ("C", "H"):
+                if elem in expected_elements:
+                    expected_elements.pop(expected_elements.index(elem))
+                    expected_elements.insert(0, elem)
+
+        if any(elem not in CHEMICAL_SYMBOLS for elem in elements):
+            raise ValueError(
+                f"Cannot use unknown chemical symbols {[elem for elem in elements if elem not in CHEMICAL_SYMBOLS]} in {field.name!r}"
+            )
+
+        if "1" in numbers:
+            raise ValueError(
+                "Must omit proportion '1' from formula {v} in {field.name!r}"
+            )
+        if expected_elements != elements:
+            order = "Hill" if field == "chemical_formula_hill" else "alphabetical"
+            raise ValueError(
+                f"Elements in {field.name!r} must appear in {order} order: {expected_elements} not {elements}."
+            )
+
+        return v
+
+    @validator("chemical_formula_anonymous")
+    def check_anonymous_formula(cls, v):
+        elements = tuple(re.findall(r"[A-Z][a-z]*", v))
+        numbers = [n.strip() for n in re.split(r"[A-Z][a-z]*", v)[1:]]
+
+        if any(n == "1" for n in numbers):
+            raise ValueError(
+                "'chemical_formula_anonymous' {v} must omit proportion '1'"
+            )
+
+        if not all(not n or n.isdigit() for n in numbers):
+            raise ValueError(
+                f"'chemical_formula_anonymous' {v} contained non-digit in proportion"
+            )
+
+        try:
+            numbers = [int(n) if n.strip() else 1 for n in numbers]
```

Comment:
If you are sure about the previous comment, `strip` should be removable here.
```suggestion
            numbers = [int(n) if n else 1 for n in numbers]
```

Reply:
You're right, its especially useless after committing your `.strip()` suggestion a few lines further up.